src/github.com/matrix-org/rageshake/submit.go

```diff
@@ -41,10 +41,12 @@ type submitServer struct {
 }
 
 type payload struct {
-	Text      string     `json:"text"`
-	Version   string     `json:"version"`
-	UserAgent string     `json:"user_agent"`
-	Logs      []logEntry `json:"logs"`
+	Text      string            `json:"text"`
+	AppName   string            `json:"app"`
+	Version   string            `json:"version"`
+	UserAgent string            `json:"user_agent"`
+	Logs      []logEntry        `json:"logs"`
+	Data      map[string]string `json:"data"`
```

Comment:
> neat way to achieve this

This is Go. Of course there isn't a neat way. You may be able to abuse `json.RawMessage` and then just read the literal `[]byte` given it's JSON. E.g: https://play.golang.org/p/-b3BpeSBgf